spec/services/subscription_variants_service_spec.rb

```diff
@@ -0,0 +1,130 @@
+require "spec_helper"
+
+describe SubscriptionVariantsService do
+  describe "variant eligibility for subscription" do
+    let!(:shop) { create(:distributor_enterprise) }
+    let!(:producer) { create(:supplier_enterprise) }
+    let!(:product) { create(:product, supplier: producer) }
+    let!(:variant) { product.variants.first }
+
+    let!(:schedule) { create(:schedule, order_cycles: [order_cycle]) }
+    let!(:subscription) { create(:subscription, shop: shop, schedule: schedule) }
+    let!(:subscription_line_item) do
+      create(:subscription_line_item, subscription: subscription, variant: variant)
+    end
+
+    let(:current_order_cycle) do
+      create(:simple_order_cycle, coordinator: shop, orders_open_at: 1.week.ago,
+                                  orders_close_at: 1.week.from_now)
+    end
+
+    let(:future_order_cycle) do
+      create(:simple_order_cycle, coordinator: shop, orders_open_at: 1.week.from_now,
+                                  orders_close_at: 2.weeks.from_now)
+    end
+
+    let(:past_order_cycle) do
+      create(:simple_order_cycle, coordinator: shop, orders_open_at: 2.weeks.ago,
+                                  orders_close_at: 1.week.ago)
+    end
+
+    let!(:order_cycle) { current_order_cycle }
+
+    context "if the shop is the supplier for the product" do
+      let!(:producer) { shop }
+
+      it "is eligible" do
+        expect(described_class.eligible_variants(shop)).to include(variant)
```

Comment:
we should have a convention for the use of described_class. if this is the way we like it, we can make it mandatory. what do you guys think?

Reply:
I'd stick to it, yes.

Reply:
Interesting, it turns out we're not using `described_class` that much:

```sh
$ git grep described_class | wc -l
30
```